libs/shared/flagd-js-core/src/lib/feature-flag.ts

```diff
@@ -0,0 +1,202 @@
+import {FlagValue} from '@openfeature/server-sdk';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/**
+ * Flagd flag configuration structure.
+ */
+export class FeatureFlag {
+  private readonly _state: string;
+  private readonly _defaultVariant: string;
+  private readonly _variants: Map<string, FlagValue>;
+  private readonly _targetingString: string;
+
+  constructor(flag: any) {
+    this._state = flag['state'];
+    this._defaultVariant = flag['defaultVariant'];
+    this._variants = new Map<string, FlagValue>(Object.entries(flag['variants']));
+    this._targetingString = JSON.stringify(flag['targeting']);
+  }
+
+  get state(): string {
+    return this._state;
+  }
+
+  get defaultVariant(): string {
+    return this._defaultVariant;
+  }
+
+  get targetingString(): string {
+    return this._targetingString;
+  }
+
+  get variants(): Map<string, FlagValue> {
+    return this._variants;
+  }
+}
+
+// flagd flag definition schema
+export const flagdSchema = {
```

Comment:
This should be its own file.

Reply:
Changed to obtain this from flagd-schema (as proposed by @toddbaert )